src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsClient/DiagnosticsClient.cs

```diff
@@ -562,18 +577,37 @@ internal static bool ValidateResponseMessage(IpcMessage responseMessage, string
                                 return false;
                             }
                             throw new UnsupportedCommandException($"{operationName} failed - Command is not supported.");
+
                         case (uint)DiagnosticsIpcError.ProfilerAlreadyActive:
                             throw new ProfilerAlreadyActiveException($"{operationName} failed - A profiler is already loaded.");
+
                         case (uint)DiagnosticsIpcError.InvalidArgument:
                             if (options.HasFlag(ValidateResponseOptions.InvalidArgumentIsRequiresSuspension))
                             {
                                 throw new ServerErrorException($"{operationName} failed - The runtime must be suspended for this command.");
                             }
                             throw new UnsupportedCommandException($"{operationName} failed - Invalid command argument.");
+
+                        case (uint)DiagnosticsIpcError.NotSupported:
```

Comment:
Is there a useful difference between 'NotSupported' and 'UnknownCommand'? 

All the other exceptions thrown here derive from the ServerErrorException which makes it convenient for callers to catch that type. Ideally we shouldn't throw something outside that hierarchy unless we think it is important for this error not to be treated similarly.

Reply:
Mono returns (always has) this error for any of the generate core dump commands.  I wanted to return better error message for it.

Reply:
From the caller's perspective do they need to be concerned about the difference between Mono's error code and CoreCLR's? I'm guessing they have no reason to care in which case we could map both errors to the identical UnknownCommandException?

Reply:
> Are you concerned that the NotSupported error is throwing a NotSupportedException exception instead of ServerErrorException like it did before my change

Yeah, that one : )